tracer/src/Datadog.Trace/AppSec/Rasp/RaspModule.cs

```diff
@@ -177,4 +179,26 @@ private static void SendStack(Span rootSpan, string id)
             rootSpan.Context.TraceContext.AddStackTraceElement(stack, Security.Instance.Settings.MaxStackTraces);
         }
     }
+
+    internal static void OnCommandInjection(string fileName, string argumentLine, Collection<string>? argumentList, bool useShellExecute)
```

Comment:
FYI, this code (or wherever is safest) _needs_ to call the `ProcessHelpers.ShouldTraceProcessStart()` method. We invoke processes during startup to e.g. read info about the environment, so we have to make sure _not_ to invoke _any_ code that could cause things to initialize, i.e. calling `Security.Instance` or `Tracer.Instance`, otherwise we end up recursively initializing which causes "issues" 😅

Reply:
Yes, we check that on ProcessStartIntegration.cs, which is actually calling the vulnerabilityManager, which calls the RaspModule.